csrc/host_ir/executor.cpp

```diff
@@ -7,13 +7,157 @@
 // clang-format on
 
 #include <dynamic_transform.h>
+#include <fusion_profiler.h>
 #include <host_ir/executor.h>
+#include <instrumentation.h>
 #include <ir/utils.h>
+#include <multidevice/communication.h>
+#include <multidevice/lower_communication.h>
+#include <multidevice/utils.h>
+#include <options.h>
+#include <runtime/allocations.h>
+#include <runtime/executor_dispatch.h>
 #include <runtime/executor_kernel_arg.h>
 #include <runtime/fusion_kernel_runtime.h>
 
 namespace nvfuser {
 
+HostIrExecutor::HostIrExecutor(
+    int64_t fusion_id,
+    int64_t concrete_id,
+    int64_t runtime_id,
+    int64_t group_id)
+    : ExecutorAbstract(fusion_id, concrete_id, runtime_id, group_id),
+      communicator_(&Communicator::getInstance()) {}
+
+bool HostIrExecutor::supported(Fusion* fusion) {
+  FUSER_PERF_SCOPE("HostIrExecutor::supported");
+  if (fusion->isA<hir::HostIrContainer>()) {
```

Comment:
Do you expect ::supported to be called both pre and post compilation? Why is post needed? I presume after compilation we get a list/DAG of ExecutorDispatch so we can simply call ExecutorDispatch::run for each.

Reply:
I'm having trouble linking your comment to this line of code. I believe this is here because some of the tests are directly creating a HostIrContainer, and sending that to execution. So, for those cases I wanted to make sure the HostIrExecutor was selected. Maybe I misunderstood and this isn't how the execution path should go, though all the tests are working now (barring maybe one that's rerunning after merging main in).

Reply:
> I believe this is here because some of the tests are directly creating a HostIrContainer, and sending that to execution.

This is normal. However, I'd expect HostIrEvaluator to pick up that HostIrContainer, and give HostUnit in PostOnStream to ExecutorAbstract. That HostUnit, IIRC, contains a plain fusion instead of another HostIrContainer. 

(great to know tests are passing; happy with addressing this in a separate PR)

cc @samnordmann

Reply:
I think you're right Jingyue, I must have been mistaken when adding this, it looks like no tests rely on it. Must have been my misinterpretation of what was happening in the host_ir tests. Will revert and double check the tests pass.